features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/utils/LegacyDrawableToColorMapper.kt

```diff
@@ -10,27 +10,42 @@ package com.datadog.android.sessionreplay.utils
 import android.annotation.SuppressLint
 import android.graphics.Paint
 import android.graphics.PorterDuffColorFilter
+import android.graphics.drawable.BitmapDrawable
 import android.graphics.drawable.ColorDrawable
 import android.graphics.drawable.Drawable
 import android.graphics.drawable.GradientDrawable
 import android.graphics.drawable.InsetDrawable
 import android.graphics.drawable.LayerDrawable
 import android.graphics.drawable.RippleDrawable
+import android.graphics.drawable.ShapeDrawable
+import android.graphics.drawable.VectorDrawable
 import com.datadog.android.api.InternalLogger
 
 /**
  * Drawable utility object needed in the Session Replay Wireframe Mappers.
  * This class is meant for internal usage so please use it carefully as it might change in time.
  */
-open class LegacyDrawableToColorMapper : DrawableToColorMapper {
+open class LegacyDrawableToColorMapper(
+    private val extensionMappers: List<DrawableToColorMapper> = emptyList()
+) : DrawableToColorMapper {
 
     override fun mapDrawableToColor(drawable: Drawable, internalLogger: InternalLogger): Int? {
+        // First check if extension mappers can resolve the drawable
+        extensionMappers.forEach {
+            val result = it.mapDrawableToColor(drawable, internalLogger)
+            if (result != null) {
+                return result
```

Comment:
minor: to avoid confusion with the return target (`forEach` vs `mapDrawableToColor`) maybe it is better to use `firstOrNull` and then do null check?

Reply:
`firstOrNull` returns the `extensionMapper` which meets the condition instead of produced value, then we need to call the function again?
The only operator seems work here is `firstNotNullOfOrNull` but you told me this API is complex to understand ?

Reply:
`firstNotNullOf ` is not safe to call if all the produced value are null, it can throw `NoSuchElementException`, that's why there is  an API called `firstNotNullOfOrNull`